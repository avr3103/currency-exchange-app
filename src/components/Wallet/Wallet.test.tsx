import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Wallet } from './index';

describe('Wallet component', () => {
  const changeValue = jest.fn();
  const changeCurrency = jest.fn();

  const props = {
    inputValue: 100,
    currency:'USD',
    walletName: 'walletFrom',
    amount:1000,
    currencySymbol:'$',
    onChangeAmount:changeValue,
    onChangeCurrency: changeCurrency
  };

  it('renders Wallet', () => {
    const { asFragment } = render(<Wallet {...props}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('changes Wallet currency option', () => {
    render(<Wallet {...props}/>);

    const select = screen.getByRole('combobox');
    fireEvent.select(select, {target: {value: 'GBP'}});
    expect(select).toHaveProperty('value', 'GBP');
  });

  it('input sign is exist', () => {
    render(<Wallet {...props}/>);

    expect(screen.getByText(`-`)).toBeInTheDocument();
  })
});