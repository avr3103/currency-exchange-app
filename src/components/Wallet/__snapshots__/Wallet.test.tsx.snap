// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet component renders Wallet 1`] = `
<DocumentFragment>
  <div
    class="wallet"
  >
    <div
      class="row"
    >
      <select
        class="wallet-currency"
        name="walletFrom"
      >
        <option
          value="EUR"
        >
          EUR
        </option>
        <option
          value="USD"
        >
          USD
        </option>
        <option
          value="GBP"
        >
          GBP
        </option>
      </select>
      <div>
        <span
          class="wallet-operation-sign"
        >
          -
        </span>
        <input
          class="wallet-input"
          name="walletFrom"
          type="number"
          value="100"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <span
        class="wallet-amount"
      >
        You have: 1000$
      </span>
    </div>
  </div>
</DocumentFragment>
`;
